### PR TITLE
More filter options: regions and germplasm

### PR DESCRIPTION
**Pull request by Jiu9Shen**
## Metadata

 - Issue #

- [ ] I feel this PR is ready to be merged.
- [ ] This PR is dependent upon [PR #/ nothing]

Documentation:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Testing?

- [ ] I tested on a generic Tripal Site
- [ ] I tested on a KnowPulse Clone
- [ ] This PR includes automated testing

### CODE REVIEW

File: includes/vcf_filter.form.inc [hunk 1]
```diff
@@ -87,16 +88,57 @@ function vcf_filter_form($form, &$form_state) {
...
   $form['s2'] = array(
     '#type' => 'fieldset',
     '#title' => 'Specify filter criteria.',
     '#tree' => TRUE,
+    '#suffix' => '</div>',
+  );
+
+  $form['s2']['restrict_dataset'] = array(
+    '#type' => 'fieldset',
+    '#title' => 'Restrict dataset',
```
Comment (laceysanderson) [inline includes/vcf_filter.form.inc]:
```suggestion
    '#title' => 'Restrict dataset to specific germplasm or regions',
```
Maybe be more specific in the fieldset title if it's going to be closed by default?

File: includes/vcf_filter.form.inc [hunk 2]
```diff
@@ -87,16 +88,57 @@ function vcf_filter_form($form, &$form_state) {
...
+    '#tree' => TRUE,
+    '#collapsible' => TRUE,
+    '#collapsed' => TRUE,
   );
+  
+    $form['s2']['restrict_dataset']['regions'] = array(
+    '#type' => 'textarea',
+    '#title' => 'Regions',
+    '#description' => 'Only include sites in these regions. <strong>Please leave it blank if filter by regions is not required.</strong> <br>Region input format: Chrom:position-lower-bound..position-upper-bound (e.g., Chr1:111100..222200). ',
```
Comment (laceysanderson) [inline includes/vcf_filter.form.inc]:
```suggestion
    '#description' => 'Only include sites in specific regions. For example, if you want to include all genotypic data for a QTL on Chr1 from 5555 to 6666 then you would enter Chr1:5555..6666 <br>Region input format: Chrom:position-lower-bound..position-upper-bound (e.g., Chr1:111100..222200). ',
```

File: includes/vcf_filter.form.inc [hunk 3]
```diff
@@ -87,16 +88,57 @@ function vcf_filter_form($form, &$form_state) {
...
+    '#value' => user_example_germplasm_names_file(),
+    '#disable' => TRUE,
+    '#description' => 'All germplasm (individuals) in selected file. Please copy the germplasm that need to be kept to textarea under.',
+    '#rows' => 2,
+    );
+    
+  $form['s2']['restrict_dataset']['germplasm_select'] = array(
+    '#type' => 'textarea',
+    '#title' => 'Keep these Germplasm',
+    '#description' => 'Only include these germplasm (individuals) in export file. <strong>Please leave it blank if filter by germplasm is not required.</strong>',
```
Comment (laceysanderson) [inline includes/vcf_filter.form.inc]:
```suggestion
    '#description' => 'Only include these germplasm (individuals) in export file. Each germplasm name should be on it's own line and must match exactly the names in the file chosen (names are shown above).',
```

**Comment by Jiu9Shen** [comment]
Loading time after choose a file depends on the size of this file, which is about 3 seconds for a 10G VCF file. So, it's a problem when file size is bigger than 30G. Trying to find a solution of "how to terminate grep command after first match".

File: includes/vcf_filter.form.inc [hunk 4]
```diff
@@ -87,17 +89,80 @@ function vcf_filter_form($form, &$form_state) {
...
+    if (!empty($curr_file['specific_germplasm'])) {
+      $value_4_germplasm_availalbe = $curr_file['specific_germplasm'];
+    }
+    if (!empty($curr_file['chromosome_format'])) {
+      $value_4_attributes_placeholder = trim($curr_file['chromosome_format'] . ':111111..222222');
+      $value_4_attributes_placeholder = array('placeholder' => $value_4_attributes_placeholder);
+    }    
+    
+  }
+
```
Comment (laceysanderson) [inline includes/vcf_filter.form.inc]: I worry the double textareas will be confusing. Maybe a warning will help until we can think of a better solution.
```suggestion

    $form['s2']['restrict_dataset']['help'] = array(
    '#type' => 'markup',
    '#markup' => '<div class="messages warning">You must <strong>copy</strong> the germplasm you want to keep into the <strong>Keep these Germplasm</strong> or your file will contain all germplasm.</div>'
  );
  
```

**Pull request merged**